http/src/handler.rs

```diff
@@ -28,6 +30,7 @@ pub struct ServerHandler<M: Metadata = (), S: Middleware<M> = middleware::Noop>
 	health_api: Option<(String, String)>,
 	max_request_body_size: usize,
 	keep_alive: bool,
+	response_middleware: Arc<dyn ResponseMiddleware>,
```

Comment:
What do you think of making this optional? This way we'll be able to not impact the existing interface of ServerHandler creation. Or just setting it to the noop in the server handler consturctor, and then relying on the `response_middleware` to set it when needed. Or is new not the main method of creating the handler?